CoreFoundation/Base.subproj/CFPlatform.c

```diff
@@ -180,6 +180,12 @@ const char *_CFProcessPath(void) {
     }
     return __CFProcessPath;
 }
+
+#else
+
+Boolean _CFIsMainThread(void) {
+    return pthread_main_np() == 1;
+}
```

Comment:
It also works for Windows since there is a `#define pthread_main_np _NS_pthread_main_np` wrapped inside a `#if DEPLOYMENT_TARGET_WINDOWS`